Sources/Tracking/Service/HttpRequestRunner.swift

```diff
@@ -57,6 +58,33 @@ internal class UrlRequestHttpRequestRunner: HttpRequestRunner {
         }.resume()
     }
 
+    public func downloadFile(url: URL, onComplete: @escaping (URL?) -> Void) {
+        guard let documentsDirectoryURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first
+        else { return }
+
+        session.downloadTask(with: url) { tempLocation, _, _ in
+            guard let tempLocation = tempLocation else {
+                return onComplete(nil)
+            }
+
+            // note make sure that the file name below is unique to not overwrite previous file.
+            var uniqueFileName = String.random
+            if let fileExtension = url.absoluteString.fileExtension {
```

Comment:
file metadata is also available as a `Content-Type` header if we wanted to be particularly robust, but extension is probably fine and saves us having to parse the MIME Type